[Contact] nightly build is failing

Since 6 days the nightly build in the Contact is failing, unfortunately only with GCC!

@loumalouomega do these test use the elimination B&S?
=> #8853 (FYI @ashishdarekar) was merged on that day, maybe sth went wrong? I reviewed it myself and it seemed fine to me.

~~~
Running ContactStructuralMechanicsApplication tests
.............F.........F...FF..F....................................................
======================================================================
FAIL: test_execution (SmallTests.PenaltyFrictionlessHyperSimplePatchFrictionalTestContact)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/__w/Kratos/Kratos/bin/Release/applications/ContactStructuralMechanicsApplication/tests/contact_structural_mechanics_test_factory.py", line 76, in test_execution
    self.test.RunSolutionLoop()
  File "/__w/Kratos/Kratos/bin/Release/KratosMultiphysics/analysis_stage.py", line 67, in RunSolutionLoop
    self.FinalizeSolutionStep()
  File "/__w/Kratos/Kratos/bin/Release/KratosMultiphysics/analysis_stage.py", line 149, in FinalizeSolutionStep
    process.ExecuteFinalizeSolutionStep()
  File "/__w/Kratos/Kratos/bin/Release/KratosMultiphysics/from_json_check_result_process.py", line 54, in ExecuteFinalizeSolutionStep
    self.assertTrue(self.process.IsCorrectResult(), "Results do not coincide with the JSON reference\n" + self.process.GetErrorMessage())
AssertionError: False is not true : Results do not coincide with the JSON reference
Error checking for variable DISPLACEMENT_X, results:
Node 3 1.040834e-17!=0.005534698; rel_tol=1e-06, abs_tol=0.001
Error checking for variable DISPLACEMENT_X, results:
Node 4 8.673617e-18!=-0.005534698; rel_tol=1e-06, abs_tol=0.001
Error checking for variable DISPLACEMENT_X, results:
Node 5 -4.015288e-05!=-0.005574027; rel_tol=1e-06, abs_tol=0.001
Error checking for variable DISPLACEMENT_X, results:
Node 6 4.015288e-05!=0.005574027; rel_tol=1e-06, abs_tol=0.001
Error checking for variable DISPLACEMENT_Y, results:
Node 3 -1.387779e-17!=-0.01941737; rel_tol=1e-06, abs_tol=0.001
Error checking for variable DISPLACEMENT_Y, results:
Node 5 -0.03985925!=-0.02046151; rel_tol=1e-06, abs_tol=0.001
Error checking for variable DISPLACEMENT_Y, results:
Node 4 6.938894e-18!=-0.01941737; rel_tol=1e-06, abs_tol=0.001
Error checking for variable DISPLACEMENT_Y, results:
Node 6 -0.03985925!=-0.02046151; rel_tol=1e-06, abs_tol=0.001


======================================================================
FAIL: test_execution (SmallTests.ALMPerfectStickHyperSimplePatchFrictionalTestContact)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/__w/Kratos/Kratos/bin/Release/applications/ContactStructuralMechanicsApplication/tests/contact_structural_mechanics_test_factory.py", line 76, in test_execution
    self.test.RunSolutionLoop()
  File "/__w/Kratos/Kratos/bin/Release/KratosMultiphysics/analysis_stage.py", line 67, in RunSolutionLoop
    self.FinalizeSolutionStep()
  File "/__w/Kratos/Kratos/bin/Release/KratosMultiphysics/analysis_stage.py", line 149, in FinalizeSolutionStep
    process.ExecuteFinalizeSolutionStep()
  File "/__w/Kratos/Kratos/bin/Release/KratosMultiphysics/from_json_check_result_process.py", line 54, in ExecuteFinalizeSolutionStep
    self.assertTrue(self.process.IsCorrectResult(), "Results do not coincide with the JSON reference\n" + self.process.GetErrorMessage())
AssertionError: False is not true : Results do not coincide with the JSON reference
Error checking for variable DISPLACEMENT_X, results:
Node 3 3.885781e-16!=0.01422514; rel_tol=1e-06, abs_tol=0.001
Error checking for variable DISPLACEMENT_X, results:
Node 5 -0.0002007644!=-0.01422514; rel_tol=1e-06, abs_tol=0.001
Error checking for variable DISPLACEMENT_X, results:
Node 4 6.661338e-16!=-0.01422514; rel_tol=1e-06, abs_tol=0.001
Error checking for variable DISPLACEMENT_X, results:
Node 6 0.0002007644!=0.01422514; rel_tol=1e-06, abs_tol=0.001
Error checking for variable DISPLACEMENT_Y, results:
Node 3 -2.775558e-16!=-0.109236; rel_tol=1e-06, abs_tol=0.001
Error checking for variable DISPLACEMENT_Y, results:
Node 5 -0.1992963!=-0.110236; rel_tol=1e-06, abs_tol=0.001
Error checking for variable DISPLACEMENT_Y, results:
Node 4 2.498002e-16!=-0.109236; rel_tol=1e-06, abs_tol=0.001
Error checking for variable DISPLACEMENT_Y, results:
Node 6 -0.1992963!=-0.110236; rel_tol=1e-06, abs_tol=0.001


======================================================================
FAIL: test_execution (SmallTests.PenaltyNoFrictionHyperSimplePatchFrictionalTestContact)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/__w/Kratos/Kratos/bin/Release/applications/ContactStructuralMechanicsApplication/tests/contact_structural_mechanics_test_factory.py", line 76, in test_execution
    self.test.RunSolutionLoop()
  File "/__w/Kratos/Kratos/bin/Release/KratosMultiphysics/analysis_stage.py", line 67, in RunSolutionLoop
    self.FinalizeSolutionStep()
  File "/__w/Kratos/Kratos/bin/Release/KratosMultiphysics/analysis_stage.py", line 149, in FinalizeSolutionStep
    process.ExecuteFinalizeSolutionStep()
  File "/__w/Kratos/Kratos/bin/Release/KratosMultiphysics/from_json_check_result_process.py", line 54, in ExecuteFinalizeSolutionStep
    self.assertTrue(self.process.IsCorrectResult(), "Results do not coincide with the JSON reference\n" + self.process.GetErrorMessage())
AssertionError: False is not true : Results do not coincide with the JSON reference
Error checking for variable DISPLACEMENT_X, results:
Node 3 0.01210697!=0.005540362; rel_tol=1e-06, abs_tol=0.001
Error checking for variable DISPLACEMENT_X, results:
Node 5 -0.01015662!=-0.00557969; rel_tol=1e-06, abs_tol=0.001
Error checking for variable DISPLACEMENT_X, results:
Node 4 0.0007751472!=-0.005540362; rel_tol=1e-06, abs_tol=0.001
Error checking for variable DISPLACEMENT_X, results:
Node 6 0.001253829!=0.00557969; rel_tol=1e-06, abs_tol=0.001
Error checking for variable DISPLACEMENT_Y, results:
Node 3 -0.02638685!=-0.01943724; rel_tol=1e-06, abs_tol=0.001
Error checking for variable DISPLACEMENT_Y, results:
Node 5 -0.01549879!=-0.02044166; rel_tol=1e-06, abs_tol=0.001
Error checking for variable DISPLACEMENT_Y, results:
Node 4 -0.01336857!=-0.01943724; rel_tol=1e-06, abs_tol=0.001
Error checking for variable DISPLACEMENT_Y, results:
Node 6 -0.02450449!=-0.02044166; rel_tol=1e-06, abs_tol=0.001


======================================================================
FAIL: test_execution (SmallTests.PenaltyPerfectStickHyperSimplePatchFrictionalTestContact)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/__w/Kratos/Kratos/bin/Release/applications/ContactStructuralMechanicsApplication/tests/contact_structural_mechanics_test_factory.py", line 76, in test_execution
    self.test.RunSolutionLoop()
  File "/__w/Kratos/Kratos/bin/Release/KratosMultiphysics/analysis_stage.py", line 67, in RunSolutionLoop
    self.FinalizeSolutionStep()
  File "/__w/Kratos/Kratos/bin/Release/KratosMultiphysics/analysis_stage.py", line 149, in FinalizeSolutionStep
    process.ExecuteFinalizeSolutionStep()
  File "/__w/Kratos/Kratos/bin/Release/KratosMultiphysics/from_json_check_result_process.py", line 54, in ExecuteFinalizeSolutionStep
    self.assertTrue(self.process.IsCorrectResult(), "Results do not coincide with the JSON reference\n" + self.process.GetErrorMessage())
AssertionError: False is not true : Results do not coincide with the JSON reference
Error checking for variable DISPLACEMENT_X, results:
Node 6 0.007265317!=0.00517577; rel_tol=1e-06, abs_tol=0.001
Error checking for variable DISPLACEMENT_X, results:
Node 3 0.02019854!=1.651197e-15; rel_tol=1e-06, abs_tol=0.001
Error checking for variable DISPLACEMENT_X, results:
Node 4 0.01823679!=1.674102e-15; rel_tol=1e-06, abs_tol=0.001
Error checking for variable DISPLACEMENT_Y, results:
Node 5 -0.009750023!=-0.02185751; rel_tol=1e-06, abs_tol=0.001
Error checking for variable DISPLACEMENT_Y, results:
Node 3 -0.0545731!=-0.02085332; rel_tol=1e-06, abs_tol=0.001
Error checking for variable DISPLACEMENT_Y, results:
Node 6 -0.01662686!=-0.02185751; rel_tol=1e-06, abs_tol=0.001
Error checking for variable DISPLACEMENT_Y, results:
Node 4 -0.008786505!=-0.02085332; rel_tol=1e-06, abs_tol=0.001


======================================================================
FAIL: test_execution (SmallTests.PenaltyHyperSimplePatchFrictionalStickTestContact)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/__w/Kratos/Kratos/bin/Release/applications/ContactStructuralMechanicsApplication/tests/contact_structural_mechanics_test_factory.py", line 76, in test_execution
    self.test.RunSolutionLoop()
  File "/__w/Kratos/Kratos/bin/Release/KratosMultiphysics/analysis_stage.py", line 67, in RunSolutionLoop
    self.FinalizeSolutionStep()
  File "/__w/Kratos/Kratos/bin/Release/KratosMultiphysics/analysis_stage.py", line 149, in FinalizeSolutionStep
    process.ExecuteFinalizeSolutionStep()
  File "/__w/Kratos/Kratos/bin/Release/KratosMultiphysics/from_json_check_result_process.py", line 54, in ExecuteFinalizeSolutionStep
    self.assertTrue(self.process.IsCorrectResult(), "Results do not coincide with the JSON reference\n" + self.process.GetErrorMessage())
AssertionError: False is not true : Results do not coincide with the JSON reference
Error checking for variable DISPLACEMENT_X, results:
Node 5 -0.004935556!=-0.001500374; rel_tol=1e-06, abs_tol=0.001
Error checking for variable DISPLACEMENT_X, results:
Node 6 0.006248999!=0.002052505; rel_tol=1e-06, abs_tol=0.001
Error checking for variable DISPLACEMENT_Y, results:
Node 5 -0.02372439!=-0.037413; rel_tol=1e-06, abs_tol=0.001
Error checking for variable DISPLACEMENT_Y, results:
Node 6 -0.02506053!=-0.0400616; rel_tol=1e-06, abs_tol=0.001
Error checking for variable DISPLACEMENT_Y, results:
Node 3 -0.02365966!=-0.006956304; rel_tol=1e-06, abs_tol=0.001
Error checking for variable DISPLACEMENT_Y, results:
Node 4 -0.02236035!=-0.00695614; rel_tol=1e-06, abs_tol=0.001


----------------------------------------------------------------------
~~~

--- a/applications/ContactStructuralMechanicsApplication/custom_conditions/mortar_contact_condition.cpp
+++ b/applications/ContactStructuralMechanicsApplication/custom_conditions/mortar_contact_condition.cpp
@@ -338,7 +338,7 @@ void MortarContactCondition<TDim, TNumNodes, TFrictional, TNormalVariation, TNum
 
     const double geometry_area = r_slave_geometry.Area();
     if (is_inside && ((integration_area/geometry_area) > 1.0e-5)) {
-        IntegrationMethod this_integration_method = GetIntegrationMethod();
+        const IntegrationMethod this_integration_method = GetIntegrationMethod();
 
         // Initialize general variables for the current master element
         general_variables.Initialize();
@@ -353,10 +353,10 @@ void MortarContactCondition<TDim, TNumNodes, TFrictional, TNormalVariation, TNum
 
         const bool dual_LM =  DerivativesUtilitiesType::CalculateAeAndDeltaAe(r_slave_geometry, r_normal_slave, r_master_geometry, derivative_data, general_variables, consider_normal_variation, conditions_points_slave, this_integration_method, GetAxisymmetricCoefficient(general_variables));
 
+        PointerVector< PointType > points_array(TDim); // The points are stored as local coordinates, we calculate the global coordinates of this points
+        array_1d<BelongType, TDim> belong_array;
+        PointType global_point;
         for (IndexType i_geom = 0; i_geom < conditions_points_slave.size(); ++i_geom) {
-            PointerVector< PointType > points_array(TDim); // The points are stored as local coordinates, we calculate the global coordinates of this points
-            array_1d<BelongType, TDim> belong_array;
-            PointType global_point;
             for (IndexType i_node = 0; i_node < TDim; ++i_node) {
                 r_slave_geometry.GlobalCoordinates(global_point, conditions_points_slave[i_geom][i_node]);
                 points_array(i_node) = Kratos::make_shared<PointType>(PointType(global_point));

--- a/applications/ContactStructuralMechanicsApplication/custom_utilities/derivatives_utilities.cpp
+++ b/applications/ContactStructuralMechanicsApplication/custom_utilities/derivatives_utilities.cpp
@@ -662,13 +662,19 @@ void DerivativesUtilities<TDim, TNumNodes, TFrictional, TNormalVariation, TNumNo
                 double DeltaXi1 = 0.0, DeltaXi2 = 0.0;
 
                 for (IndexType i_mortar_node = 0; i_mortar_node < TNumNodes; ++i_mortar_node) {
-                    // Compute DeltaXi on the slave side if point is inside geometry (nodes of geometry excluded)
-                    if (rSlaveGeometry.IsInside( projected_in_slave[i_mortar_node].Coordinates(), aux_point_slave.Coordinates() ) && (norm_2(projected_in_slave[i_mortar_node].Coordinates() - rSlaveGeometry[0].Coordinates()) > ZeroTolerance) && (norm_2(projected_in_slave[i_mortar_node].Coordinates() - rSlaveGeometry[1].Coordinates()) > ZeroTolerance))
+                    // Compute DeltaXi on the slave side if point is inside geometry (nodes off geometry excluded)
+                    if (rSlaveGeometry.IsInside( projected_in_slave[i_mortar_node].Coordinates(), aux_point_slave.Coordinates() ) && (norm_2(projected_in_slave[i_mortar_node].Coordinates() - rSlaveGeometry[0].Coordinates()) > ZeroTolerance) && (norm_2(projected_in_slave[i_mortar_node].Coordinates() - rSlaveGeometry[1].Coordinates()) > ZeroTolerance)) {
                         DeltaXi_slave[i_mortar_node] = LocalDeltaSegmentN1( all_delta_normal, rSlaveNormal, rSlaveGeometry, rMasterGeometry, r_N1, r_DNDe1, i_mortar_node, i_node, i_dof, ConsiderNormalVariation );
+                    } else {
+                        DeltaXi_slave[i_mortar_node] = 0.0;
+                    }
 
-                    // Compute DeltaXi on the master side if point is inside geometry (nodes of geometry excluded)
-                    if (rMasterGeometry.IsInside( projected_in_master[i_mortar_node].Coordinates(), aux_point_master.Coordinates() ) && (norm_2(projected_in_master[i_mortar_node].Coordinates() - rMasterGeometry[0].Coordinates()) > ZeroTolerance) && (norm_2(projected_in_master[i_mortar_node].Coordinates() - rMasterGeometry[1].Coordinates()) > ZeroTolerance))
+                    // Compute DeltaXi on the master side if point is inside geometry (nodes off geometry excluded)
+                    if (rMasterGeometry.IsInside( projected_in_master[i_mortar_node].Coordinates(), aux_point_master.Coordinates() ) && (norm_2(projected_in_master[i_mortar_node].Coordinates() - rMasterGeometry[0].Coordinates()) > ZeroTolerance) && (norm_2(projected_in_master[i_mortar_node].Coordinates() - rMasterGeometry[1].Coordinates()) > ZeroTolerance)) {
                         DeltaXi_master[i_mortar_node] = LocalDeltaSegmentN2( all_delta_normal, rSlaveNormal, rSlaveGeometry, rMasterGeometry, r_N2, r_DNDe2, i_mortar_node, i_node, i_dof, ConsiderNormalVariation );
+                    } else {
+                        DeltaXi_master[i_mortar_node] = 0.0;
+                    }
                 }
 
                 // Evaluate DeltaXi1 expression

--- a/applications/ContactStructuralMechanicsApplication/tests/SmallTests.py
+++ b/applications/ContactStructuralMechanicsApplication/tests/SmallTests.py
@@ -72,7 +72,7 @@ class ALMHyperSimplePatchFrictionalSlipTestContact(TestFactory):
 class ALMHyperSimplePatchFrictionalStickTestContact(TestFactory):
     file_name = "ALM_frictional_contact_test_2D/hyper_simple_stick_patch_test"
 
-class PenaltyFrictionlessHyperSimplePatchFrictionalTestContact(TestFactory):
+class PenaltyFrictionlessHyperSimplePatchTestContact(TestFactory):
     file_name = "penalty_frictionless_contact_test_2D/hyper_simple_patch_test"
 
 class PenaltyNoFrictionHyperSimplePatchFrictionalTestContact(TestFactory):

--- a/applications/ContactStructuralMechanicsApplication/tests/test_ContactStructuralMechanicsApplication.py
+++ b/applications/ContactStructuralMechanicsApplication/tests/test_ContactStructuralMechanicsApplication.py
@@ -34,8 +34,8 @@ from SmallTests import ALMHyperSimpleSlopePatchTestContact                      
 from SmallTests import ALMThreeDSimplestPatchMatchingTestContact                   as TALMThreeDSimplestPatchMatchingTestContact
 
 # Penalty frictionless tests
-from SmallTests import PenaltyFrictionlessHyperSimplePatchFrictionalTestContact as TPenaltyFrictionlessHyperSimplePatchFrictionalTestContact
-from SmallTests import PenaltyThreeDSimplestPatchMatchingTestContact            as TPenaltyThreeDSimplestPatchMatchingTestContact
+from SmallTests import PenaltyFrictionlessHyperSimplePatchTestContact as TPenaltyFrictionlessHyperSimplePatchTestContact
+from SmallTests import PenaltyThreeDSimplestPatchMatchingTestContact  as TPenaltyThreeDSimplestPatchMatchingTestContact
 
 # Components ALM frictionless tests
 from SmallTests import ComponentsALMHyperSimpleTrianglePatchTestContact                      as TComponentsALMHyperSimpleTrianglePatchTestContact
@@ -211,7 +211,7 @@ def AssembleTestSuites():
     smallSuite.addTest(TALMThreeDSimplestPatchMatchingTestContact('test_execution'))
 
     # Penalty frictionless tests
-    smallSuite.addTest(TPenaltyFrictionlessHyperSimplePatchFrictionalTestContact('test_execution'))
+    smallSuite.addTest(TPenaltyFrictionlessHyperSimplePatchTestContact('test_execution'))
     smallSuite.addTest(TPenaltyThreeDSimplestPatchMatchingTestContact('test_execution'))
 
     # Components ALM frictionless tests
@@ -425,7 +425,7 @@ def AssembleTestSuites():
             #TALMThreeDPatchComplexGeomTestContact,
             #TALMTThreeDPatchMatchingTestContact,
             #TALMThreeDPatchNotMatchingTestContact,
-            #TPenaltyFrictionlessHyperSimplePatchFrictionalTestContact,
+            #TPenaltyFrictionlessHyperSimplePatchTestContact,
             #TPenaltyThreeDSimplestPatchMatchingTestContact,
             #TComponentsALMHyperSimpleTrianglePatchTestContact,
             #TComponentsALMHyperSimplePatchTestContact,

